File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxcon = 0
        count=0
        pt2=0

        while(pt2 < len(nums)):
            if nums[pt2] == 1:
                pt2 = pt2+1
                count = count+1
            elif nums[pt2] == 0:  
                count = 0
                pt2=pt2+1
            maxcon = max(count,maxcon)
        return maxcon


            

            

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
         int newwindow = 0;
         int maxcount =0;
         int count=0;
        for(int i=0;i<nums.length;i++)
        {
            if (nums[i] == 1)
            {
               count++;
            }
            else if (nums[i] != 1)
            {
                newwindow = i;
                maxcount = Math.max(count,maxcount);
                count=0;
                 
            }
        }

        maxcount = Math.max(count,maxcount);

        return maxcount;
    }

}